src/main/java/com/alipay/remoting/rpc/RpcInvokeCallbackListener.java

```diff
@@ -107,101 +107,62 @@ public CallbackTask(String remoteAddress, InvokeFuture future) {
-                                    "Connection closed when invoke with callback.The address is "
-                                            + this.remoteAddress);
-                                break;
-                            case SERVER_THREADPOOL_BUSY:
-                                e = new InvokeServerBusyException(
-                                    "Server thread pool busy when invoke with callback.The address is "
-                                            + this.remoteAddress);
-                                break;
-                            case SERVER_EXCEPTION:
-                                String msg = "Server exception when invoke with callback.Please check the server log! The address is "
-                                             + this.remoteAddress;
-                                RpcResponseCommand resp = (RpcResponseCommand) response;
-                                resp.deserialize();
-                                Object ex = resp.getResponseObject();
-                                if (ex instanceof Throwable) {
-                                    e = new InvokeServerException(msg, (Throwable) ex);
-                                } else {
-                                    e = new InvokeServerException(msg);
-                                }
-                                break;
-                            default:
-                                e = new InvokeException(
-                                    "Exception caught in invocation. The address is "
-                                            + this.remoteAddress + " responseStatus:"
-                                            + response.getResponseStatus(), future.getCause());
+                    response = (ResponseCommand) future.waitResponse(0);
+                } catch (InterruptedException e) {
+                    String msg = "Exception caught when getting response from InvokeFuture. The address is "
+                                 + this.remoteAddress;
+                    logger.error(msg, e);
+                    throw new InvokeException(msg, e);
+                }
 
-                        }
-                    }
-                    callback.onException(e);
-                } catch (Throwable e) {
-                    logger
-                        .error(
-                            "Exception occurred in user defined InvokeCallback#onException() logic, The address is {}",
-                            this.remoteAddress, e);
+                if (response == null) {
+                    throw new InvokeException("Exception caught in invocation. The address is "
+                                              + this.remoteAddress + " responseStatus:"
+                                              + ResponseStatus.UNKNOWN, future.getCause());
                 }
-            } else {
+
+                Object responseObj = RpcResponseResolver.resolveResponseObject(response,
```

Comment:
resolveResponseObject will do deserialization, but deserialization should not do before biz classloader setting